packages/core/__tests__/store/language.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { renderHook, act } from "@testing-library/react-native";
+import { SupportedLanguageCode, getLanguage, useLanguage } from "../../src/store/language";
+
+const norwegianBokmål = { code: "nb", name: "Norwegian" };
+const english = { code: "en", name: "English" };
+
+describe("Language", () => {
+    it("Should always return a language, even if no language has been set", () => {
+        const { result } = renderHook(() => useLanguage());
+        expect(result.current).toBeTruthy();
+        expect(getLanguage()).toBeTruthy();
+    });
+
+    it("Should have a default language of English if no other default language is set", () => {
+        const { result } = renderHook(() => useLanguage());
+        expect(result.current.language).toMatchObject(english);
+        expect(getLanguage()).toMatchObject(english);
+    });
+
+    it("Should be possible to change the default language", () => {
+        const { result, rerender } = renderHook(() => useLanguage());
+
+        act(() => result.current.setDefaultLanguage("nb"));
+        rerender({});
+        expect(result.current.language).toMatchObject(norwegianBokmål);
+        expect(getLanguage()).toMatchObject(norwegianBokmål);
+    });
+
+    it("Should return the user selected language if it exists", () => {
+        const { result, rerender } = renderHook(() => useLanguage());
+
+        expect(result.current.language).toMatchObject(english);
+        expect(getLanguage()).toMatchObject(english);
+
+        act(() => result.current.setSelectedLanguage("nb"));
+        rerender({});
+        expect(result.current.language).toMatchObject(norwegianBokmål);
+        expect(getLanguage()).toMatchObject(norwegianBokmål);
+
+        act(() => result.current.setDefaultLanguage("en"));
+        rerender({});
+        expect(result.current.language).toMatchObject(norwegianBokmål);
+        expect(getLanguage()).toMatchObject(norwegianBokmål);
+    });
+
+    /**
+     * This test obviously does not cover all scenarios, but it should cover
+     * The most likely language for a developer to add in equinor context. If a developer adds a new
+     * language, we should support it properly and create new tests
+     */
+    it("Should only accept English and Norwegian as language choices", () => {
```

Comment:
Not sure what the best approach is here, but I figure if we want to support a new language, for example Portuguese, it should also be supported by our common screens. The common screens will probably read from this language state anyway

Reply:
I think we should not enforce languages that all apps using core supports. I think it is better to have the whole app besides the settings screen / login screen etc to support one specific language than having no support at all.
Maybe we should default to english if the internal core screens do not support it

Reply:
If we let the apps add language support as they please, it might lead to future problems. For example, one app might support `en` and `no`, while another supports `nb` and `nn`. Suddenly we decide to add norwegian bokmål support to our common screens, so we add `nb` to the list of languages our common screens support. But the first app in this example will still only see the English version of our common screens, cuz they decided to call it `no`. So we have to add convoluted logic to make sure both no and nb is understood by mad-core. Next thing you know, some PO wants to read the settings in nynorsk instead of bokmål, so we add `nn` to our common screens. The developer adding `nn` to our common screens then notices that `no` translates to bokmål, which makes him angry. He starts an uprising, leading to a civil war between the nynorskers and the bokmålers.

After the Civil War, the developer goes back to his old job, and is tasked with adding Portuguese to our common screens. While working on that issue, he notices that 3 of the apps which supports Portuguese already have used the correct language code `pt`, while another 4 of the apps have used the language code `br`, and the language name `brazilian`. 

When Portugal hears about this, they become furious, launching a devastating attack on Brazil. The war lasts for 20 years.



To prevent these kinds of issues, I believe it's better to have a single "database" of languages that we support. The apps can pick the languages they want from that list. If the language they want is not there, they can create a PR to add it to the list, and (hopefully) add the language to our common screens as well

Reply:
Damn..

Reply:
But for real, this is avoided if we use a robust internationalisation library as a foundation since all language codes are one-to-one and unique (no fighting over no nb and nn, though nb master race 💯). If we ever run into an issue where users of core are pulling their hair out for finding the correct language code, it is not because we let them add languages themselves, but rather since we are not enforcing proper standards well enough. Again, if we used one of the language packages I mentioned in some other comment, this would be no problem

Reply:
I think I prefer the syntax of react-i18next, but from the looks of it, it doesn't enforce proper use of language codes. Looks like you can name your resources whatever you want